nextstrain/cli/command/login.py

```diff
@@ -0,0 +1,153 @@
+"""
+Logs in a Nextstrain.org groups user, storing their credentials locally.
+"""
+import os
+import ast
+import sys
+import jose
+import boto3
+import getpass
+import configparser
+from pathlib import Path
+from pycognito import Cognito
+from pycognito.aws_srp import AWSSRP
+from typing import Dict
+
+base = Path(__file__).resolve().parent.parent.parent.parent
+secrets_path = base / '.nextstrain/secrets'
+config = configparser.ConfigParser()
+
+def register_parser(subparser):
+    parser = subparser.add_parser("login", help = "Login as a Nextstrain.org groups user")
+
+    register_arguments(parser)
+
+    return parser
+
+
+def register_arguments(parser):
+    parser.add_argument(
+        "--check",
+        help    = "Check credentials by logging into nextstrain.org",
+        action  = 'store_true')
+
+    parser.add_argument(
+        "--delete",
+        help    = "Deletes the secrets file containing authentication credentials",
+        action  = 'store_true')
+
+    return parser
+
+
+def run(opts):
+    if opts.delete:
+        try:
+            os.remove(secrets_path)
+            print(f"Deleted the credentials file at {secrets_path}.")
+
+        except FileNotFoundError:
+            print("No credentials found to delete!")
+
+        sys.exit(0)
+
+    if opts.check:
+        print("Checking credentials by logging into nextstrain.org...")
+        try:
+            config.read(secrets_path)
+            username = config.get('login', 'username')
+            authn_result = ast.literal_eval(config.get('login', 'authenticationresult'))
+
+        # If the requested credentials are not found in a 'login' section in the
+        # specified secrets file, generate them, and save them locally
+        except (FileNotFoundError, configparser.NoSectionError, configparser.NoOptionError):
+            login()
+            sys.exit(0)
+
+        u = Cognito(os.environ['AWS_COGNITO_USER_POOL_ID'],
+                '2vmc93kj4fiul8uv40uqge93m5',
+                username=username,
+                id_token=authn_result['IdToken'],
+                refresh_token=authn_result['RefreshToken'],
+                access_token=authn_result['AccessToken'])
+
+        try:
+            u.verify_token(authn_result['IdToken'], 'id_token', 'id')
+            u.verify_token(authn_result['AccessToken'], 'access_token', 'access')
+
+        except (jose.exceptions.JWSError, jose.exceptions.JWTError):
+            print("Invalid credentials.")
+            print()
+            login()
+            sys.exit(0)
+
+        access_token_expired = u.check_token()
+
+        if not access_token_expired:
+            print("Logged in succesfully!")
+
+        else:
+            print("Replacing expired access token...")
+            tokens['AccessToken'] = u.access_token
```

Comment:
Ah, I think we need to store all the tokens if we're going to use any of them.

Reply:
I think I was already storing them? But let me know if I missed something here.